tests/django/test_middleware.py

```diff
@@ -32,3 +32,17 @@ def test_process_view(self):
             middleware.process_view(request, view_func, [], {})
 
         mock_inject_magic_properties.assert_called_once_with(MyView)
+
+    def test_process_view_drf_style(self):
+        view_func = MyView.as_view()
```

Comment:
Is it worth adding drf as a dependency for this project now and use a DRF viewset?